docs/content/guides/rows/rows-sorting.md

```diff
@@ -0,0 +1,3432 @@
+      brand: 'Gigabox',
+      model: 'HL Mountain Frame',
+      price: 1890.90,
+      sellDate: '03/05/2023',
+      sellTime: '11:27',
+      inStock: 22,
+    },
+    {
+      brand: 'Camido',
+      model: 'Cycling Cap',
+      price: 130.10,
+      sellDate: '27/03/2023',
+      sellTime: '03:17',
+      inStock: 13,
+    },
+    {
+      brand: 'Chatterpoint',
+      model: 'Road Tire Tube',
+      price: 59,
+      sellDate: '28/08/2023',
+      sellTime: '08:01',
+      inStock: 0,
+    },
+    {
+      brand: 'Eidel',
+      model: 'HL Road Tire',
+      price: 279.99,
+      sellDate: '02/10/2023',
+      sellTime: '13:23',
+      inStock: 14,
+    },
+    {
+      brand: 'Jetpulse',
+      model: 'Racing Socks',
+      price: 30,
+      sellDate: '11/10/2023',
+      sellTime: '01:23',
+      inStock: 16,
+    },
+    {
+      brand: 'Gigabox',
+      model: 'HL Mountain Frame',
+      price: 1890.90,
+      sellDate: '03/05/2023',
+      sellTime: '11:27',
+      inStock: 18,
+    },
+    {
+      brand: 'Camido',
+      model: 'Cycling Cap',
+      price: 130.10,
+      sellDate: '27/03/2023',
+      sellTime: '03:17',
+      inStock: 3,
+    },
+    {
+      brand: 'Chatterpoint',
+      model: 'Road Tire Tube',
+      price: 59,
+      sellDate: '28/08/2023',
+      sellTime: '08:01',
+      inStock: 0,
+    },
+    {
+      brand: 'Vinte',
+      model: 'ML Road Frame-W',
+      price: 30,
+      sellDate: '11/10/2023',
+      sellTime: '01:23',
+      inStock: 2,
+    },
+    {},
+  ],
+  columns: [
+    {
+      type: 'text',
+      data: 'brand',
+    },
+    {
+      type: 'text',
+      data: 'model',
+    },
+    {
+      type: 'numeric',
+      data: 'price',
+      numericFormat: {
+        pattern: '$ 0,0.00',
+        culture: 'en-US'
+      },
+      className: 'htLeft',
+    },
+    {
+      type: 'date',
+      data: 'sellDate',
+      className: 'htRight',
+    },
+    {
+      type: 'time',
+      data: 'sellTime',
+      correctFormat: true,
+      className: 'htRight',
+    },
+    {
+      type: 'numeric',
+      data: 'inStock',
+      className: 'htCenter',
+    },
+  ],
+  height: 200,
+  fixedRowsTop: 1,
+  fixedRowsBottom: 1,
+  colHeaders: true,
+  columnSorting: true,
+  // `afterColumnSort` is a Handsontable hook: it's fired after each sorting
+  afterColumnSort() {
+    // after each sorting, take row 1 and change its index to 0
+    handsontableInstance.rowIndexMapper.moveIndexes(
+      [
+        handsontableInstance.toVisualRow(0),
+      ], 0);
+
+    // after each sorting, take row 16 and change its index to 15
+    handsontableInstance.rowIndexMapper.moveIndexes(
+      [
+        handsontableInstance.toVisualRow(15),
+      ], 15);
+  },
+  cells(row, col, prop) {
+    if (row === 0) {
+      return {
+        type: 'text',
+        className: 'htCenter',
+      };
+    }
+    if (row === 15) {
+      return {
+        type: 'numeric',
+        className: 'htCenter',
+      };
+    }
+  },
```

Comment:
```suggestion
  afterColumnSort() {
    const lastRowIndex = handsontableInstance.countRows() - 1;
    
    // after each sorting, take row 1 and change its index to 0
    handsontableInstance.rowIndexMapper
    	.moveIndexes(handsontableInstance
      	.toVisualRow(0), 0);

    // after each sorting, take row 16 and change its index to 15
    handsontableInstance.rowIndexMapper
    	.moveIndexes(handsontableInstance
      	.toVisualRow(lastRowIndex), lastRowIndex);
  },
  cells(row, col, prop) {
    const lastRowIndex = this.instance.countRows() - 1;
    
    if (row === 0) {
      return {
        type: 'text',
        className: 'htCenter',
      };
    }
    if (row === lastRowIndex) {
      return {
        type: 'numeric',
        className: 'htCenter',
      };
    }
  },
```
I changed the demo slightly, and I used API to freeze the last row based on the dataset length, not the hardcoded values. So it has a chance to work when the user starts changing the number of rows - for example.